src/dataflow/src/sink/tail.rs

```diff
@@ -76,3 +95,20 @@ pub fn tail<G>(
         });
     })
 }
+
+fn batch_upper(
+    batch: Option<&Rc<OrdValBatch<GlobalId, Row, u64, isize, usize>>>,
+) -> Option<Timestamp> {
+    match batch {
+        Some(batch) => {
+            // Break the antichain abstraction here by fetching the first element because
+            // we know there won't ever be 2.
+            let upper = match batch.desc.upper().elements().get(0) {
+                Some(upper) => upper,
+                None => return None,
+            };
+            Some(*upper)
+        }
+        None => None,
+    }
```

Comment:
I think this could possibly all be resolved down to `batch.map(|b| *b.desc.upper().elements.get(0))`, or is that wrong for some reason (Rust)?

Reply:
Had to add `.flatten().map(..)` but yes thanks!